controllers/kubevirtmachine_controller_test.go

```diff
@@ -289,6 +290,15 @@ var _ = Describe("reconcile a kubevirt machine", func() {
 		// Should expect kubevirt machine is still not ready
 		Expect(machineContext.KubevirtMachine.Status.Ready).To(BeFalse())
 		Expect(machineContext.KubevirtMachine.Spec.ProviderID).To(BeNil())
+
+		// Should have created the userdata secret
+		machineBootstrapSecretReferenceName := machineContext.Machine.Spec.Bootstrap.DataSecretName
+		machineBootstrapSecretReferenceKey := client.ObjectKey{Namespace: machineContext.Machine.GetNamespace(), Name: *machineBootstrapSecretReferenceName + "-userdata"}
+		bootstrapDataSecret := &corev1.Secret{}
+		err = fakeClient.Get(gocontext.Background(), machineBootstrapSecretReferenceKey, bootstrapDataSecret)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(bootstrapDataSecret.Data["userdata"]).To(Equal([]byte("shell-script")))
+		Expect(bootstrapDataSecret.Labels).To(Equal(map[string]string{"hello": "world"}))
```

Comment:
Please use gomega matchers for readability:
```suggestion
		Expect(err).NotTo(HaveOccurred())
		Expect(bootstrapDataSecret.Data).To(HaveKeyWithValue("userdata", "shell-script"))
		Expect(bootstrapDataSecret.Labels).To(HaveLen(1))
		Expect(bootstrapDataSecret.Labels).To(HaveKeyWithValue("hello", "world"))
```

Reply:
Thanks for the feedback @nunnatsa .